internal/controller/controllers/agent_controller.go

```diff
@@ -588,11 +594,31 @@ func (r *AgentReconciler) updateStatus(ctx context.Context, log logrus.FieldLogg
 	} else {
 		setConditionsUnknown(agent)
 	}
+	// get node
+	var node *corev1.Node
+	if funk.Contains([]models.HostStage{models.HostStageRebooting, models.HostStageJoined, models.HostStageConfiguring}, agent.Status.Progress.CurrentStage) {
+		node, err = r.getNode(ctx, agent)
+		if err != nil && !k8serrors.IsNotFound(err) {
+			r.Log.WithError(err).Errorf("agent %s/%s: failed to get node", agent.Namespace, agent.Name)
+			return ctrl.Result{}, err
+		}
+		if node != nil {
+			if isNodeReady(node) {
+				err = r.updateHostInstallProgress(ctx, h, models.HostStageDone)
+				agent.Status.Progress.CurrentStage = models.HostStageDone
+			} else {
+				err = r.updateHostInstallProgress(ctx, h, models.HostStageJoined)
+				agent.Status.Progress.CurrentStage = models.HostStageJoined
```

Comment:
Since the status/stage change logic is now separate from the CSR approval logic, maybe the status/stage calculation can be moved to a separate function ? 
Also consider to include the code [between lines 558-563](https://github.com/openshift/assisted-service/blob/3cc5fe4735485ef23b6e7dc6469157749e27379a/internal/controller/controllers/agent_controller.go#L558-L561)